src/Containers/ReactTable.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import styled from 'styled-components';
+import { useTable, Column, TableHeaderProps, TableRowProps } from 'react-table';
+import { IProfileProps } from './VendorsList/Profile';
+import { MainInterface, ResponsiveInterface } from '../Utils/BaseStyles';
+import { media, scroll } from '../Utils/Mixins';
+
+export interface IVendorsData extends IProfileProps {
+    tags?: string[];
+    createdAt?: string;
+};
+
+export interface IReactTableProps<T extends IVendorsData> extends MainInterface, ResponsiveInterface,React.HTMLAttributes<HTMLDivElement> {
+    data: T[];
+    columns: Column<T>[];
+    tableHeaderProps?: TableHeaderProps;
+    tableRowProps?: TableRowProps;
+};
+
+export const ReactTable: React.FC<IReactTableProps<any>> = ({
+    data,
+    columns,
+    tableHeaderProps,
+    tableRowProps,
+    ...props
+}): React.ReactElement => {
+    const {
+        getTableProps,
+        getTableBodyProps,
+        headerGroups,
+        rows,
+        prepareRow
+    } = useTable({
+        columns, 
+        data
+    });
+
+    return (
+        <table {...getTableProps()} {...props}>
+            <STableHead>
+                {headerGroups.map((headerGroup, index) => (
+                    <SHeadTableRow {...headerGroup.getHeaderGroupProps()} key={headerGroup.headers[index].Header?.toString()}>
+                        {headerGroup.headers.map(column => (
+                            <STableHeader {...column.getHeaderProps()} key={column.Header?.toString()} {...tableHeaderProps}>
+                                {column.render('Header')}
+                            </STableHeader>
+                        ))}
```

Comment:
It's best to keep the tree small and make a function called getColumns, that way you can wrap it with a useCallback / Memo if needed